buildrunner/steprunner/__init__.py

```diff
@@ -92,9 +92,9 @@ def run(self):
             for _task_name, _task_config in self.config.items():
                 self.log.write(f"==> Running step: {self.name}:{_task_name}\n")
                 if _task_name in TASK_MAPPINGS:
-                    if self.local_images:
+                    if self.local_images and _task_name in ["build", "run", "services"]:
                         _task_config["pull"] = False
-                    if self.platform:
+                    if self.platform and _task_name in ["build", "run"]:
```

Comment:
I'd prefer using the static class names (i.e. check after finding the _task class instead of just checking the name) instead of these (potentially variable) config names. But since this config is pretty stable, I guess it doesn't really matter.

Reply:
Updated. I left off the "services" since that is not found in the TASK_MAPPINGS.